### PR TITLE
tlp: add package

### PR DESCRIPTION
**Pull request by abbradar**
This adds tlp daemon, a set of power-saving  scripts with additional options for ThinkPads.

### CODE REVIEW

**Comment by abbradar** [comment]
Please wait with merging this; I've discovered that tpacpi-bat is not working (environment issues) -- will try to come up with a fix.

**Comment by abbradar** [comment]
Fixed; this needs https://github.com/NixOS/nixpkgs/pull/5593 now for working `tpacpi-bat`, though if one doesn't need it he can use it as-is.

**Comment by devhell** [comment]
What's the status on this?

**Comment by abbradar** [comment]
Working well for me since my last comment. Needs #5593 if you want `tpacpi-bat` working, otherwise fine as is.

**Comment by devhell** [comment]
:+1: Thanks for the info!

File: pkgs/tools/misc/tlp/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+, enableRDW ? true, networkmanager }:
+
+stdenv.mkDerivation {
+  inherit enableRDW;
+
+  name = "tlp-0.6";
+
+  src = fetchgit {
+    url = "https://github.com/linrunner/TLP";
+    rev = "dbcc09e0f1730e7b85c7387dcc06d01f1547edba";
```
Comment (iElectric) [inline pkgs/tools/misc/tlp/default.nix]: please use a tarball if you specify fixed version such as `0.6`

Comment (devhell) [inline pkgs/tools/misc/tlp/default.nix]: Yeah, shouldn't be a problem I hope using `fetchFromGitHub` and using a release such as this one: https://github.com/linrunner/TLP/releases/tag/0.6

**Comment by abbradar** [comment]
Fixed, thanks!

**Pull request merged**